packages/@sanity/form-builder/src/inputs/files/FileInput/InvalidFileWarning.tsx

```diff
@@ -0,0 +1,36 @@
+import {ResetIcon, WarningOutlineIcon} from '@sanity/icons'
+import {Card, Flex, Box, Text, Stack, Button} from '@sanity/ui'
+import React from 'react'
+import styled from 'styled-components'
+
+type Props = {
+  onClearValue?: () => void
+}
+
+const ButtonWrapper = styled(Button)`
+  width: 100%;
+`
+
+export function InvalidFileWarning({onClearValue}: Props) {
+  return (
+    <Card tone="caution" padding={4} border radius={2}>
+      <Flex gap={4} marginBottom={4}>
+        <Box>
+          <Text size={1}>
+            <WarningOutlineIcon />
+          </Text>
+        </Box>
+        <Stack space={3}>
+          <Text size={1} weight="semibold">
+            Invalid file value
+          </Text>
+          <Text size={1}>
+            The value of this field is not a valid file. Resetting this field will let you choose a
```

Comment:
Are we able to provide more info about why it's not valid and what may have caused it? Wondering if it would be better to reword it into a bit more cautious terms to the editor aware of the impact and that it might require a bit of consideration (e.g. someone might be in the middle of a schema migration and blindly deleting it might not be the best cause of action).

Reply:
Worth considering for v3 / both v2 and v3, but I think it is kind of an edge case for now.